packages/editor/src/components/table-of-contents/panel.js

```diff
@@ -49,7 +58,10 @@ function TableOfContentsPanel( { headingCount, paragraphCount, numberOfBlocks }
 					<span className="table-of-contents__title">
 						{ __( 'Document Outline' ) }
 					</span>
-					<DocumentOutline />
+					<DocumentOutline onSelect={ () => {
+						removeBlockIdHash();
```

Comment:
It's maybe unexpected to me that despite the implementation of the specific item being a link occurring within `DocumentOutlineItem`, we handle the hash removal here. I'm not sure `TableOfContents` should have that implementation awareness of `DcumentOutlineItem` as a link, and conversely I think any other use of `DocumentOutline` (a separate component from this one) should also have the same behavior of hash-stripping.

In other words, this seems like behavior which should occur in `DocumentOutlineItem`, not here.

Reply:
@aduth tried solving by externalizing as a helper in https://github.com/WordPress/gutenberg/pull/10815/commits/d5470e504762d2b2f62039234ea26ea661da2de4. this can't live in `DocumentOutlineItem` because it is also used to remove the hash after clicking on the outline title, which is in `DocumentOutline`